frontend/admin/resources/js/components/ClientProvider.tsx

```diff
@@ -103,7 +103,10 @@ export const ClientProvider: React.FC<{ client?: Client }> = ({
       logout();
       return null;
     },
-    [accessToken, refreshToken, idToken, logout, refreshTokenSet],
+    // We need a single instance of this function so the urql client can maintain its state.
+    // Otherwise, it loses count of the errors and enters a refresh loop.
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [],
```

Comment:
This seems dangerous... if we need this to be a stable function, it still needs the variables to be correct.
Which dependency is causing this to be regenerated? I'm guessing its the functions, logout and refreshTokenSet, we could look at AuthContainer to try to avoid creating new anonymous functions each time.
If that's not possible, maybe we could store these dependencies in an object created with useRef, and grab the current values whenever the getAuth callback runs?

Reply:
I'll admit this is near the limits of my React knowledge. :grin: The dependencies that are changing are the three tokens, right?  I probably don't have the best solution yet, though.  I'll dig back in but I'm worried that that the *best* solution is going to take some major refactoring in how behaviour is split between AuthContainer and ClientProvider.

Reply:
Thanks for looking into this Peter! I just pushed a commit, it accomplishes the same goal of avoiding new instances of `getAuth`, but without the code smell of `// eslint-disable-next-line react-hooks/exhaustive-deps`. My approach is based on the React docs' final suggestion for [What to do if effect deps change to often?](https://reactjs.org/docs/hooks-faq.html#what-can-i-do-if-my-effect-dependencies-change-too-often) Its slightly hacky, more hacky than a full refactor, but I'd be happy with it.

Reply:
Nice one! 🧙  I would not have thought of that any time soon.  I would love to refactor the AuthContainer/ClientProvider coupling someday but this is a very reasonable solution for now.  You're right that this is much better than the lint override.  I just updated the comment to reflect my better understanding of the problem.